lib/mix/test/mix/cli_test.exs

```diff
@@ -81,18 +81,18 @@ defmodule Mix.CLITest do
     end
   end
 
-  test "--help smoke test", context do
+  test "-h smoke test", context do
```

Comment:
I think we should keep current test names and probably add new assertion for `-h`.